smartcontracts/sandbox/contracts/OperatorRegistry.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+contract OperatorRegistry {
+    uint256 private _nextOperatorId;
+
+    mapping(uint256 => Operator) public operators;
+
+    struct Operator {
+        // an uri can be added with operator metadata
+        // name can be in metadata
+        string name;
+        address profile;
+        address node;
+        address rewards;
+    }
+
+    event OperatorRegistered(
+        uint256 indexed operatorId,
+        address indexed profile,
+        address indexed node,
+        address rewards,
+        string name
+    );
+
+    modifier onlyExistingOperator(uint256 operatorId) {
+        require(operators[operatorId].profile != address(0), "OperatorRegistry: unexistent operator");
+        _;
+    }
+
+    modifier onlyOperatorOwner(uint256 operatorId) {
+        require(operators[operatorId].profile == msg.sender, "OperatorRegistry: Not operator owner");
+        _;
+    }
+
+    function registerOperator(Operator memory operator) public {
+        require(
+            keccak256(abi.encodePacked(operator.name)) != keccak256(abi.encodePacked("")),
+            "OperatorRegistry: name can't be empty"
+        );
+
+        uint256 operatorId = _nextOperatorId++;
+
+        // Should we extract profileAddress from msg.sender?
```

Comment:
Yes, assume the caller is the owner of the profile. Third-party registration doesn't seem to be needed and may generate confusion/mistakes.

Reply:
Can we allow one profile manage multiple operators or will it be strictly one profile one operator

Reply:
I'd say no at this moment. I feel it would just add more complexity unless there is a clear requirement.  Redundancy is usually managed with using the same private key in multiple nodes with only one being the "active" and the others being "disabled".